docs/NNNN-direct-clean-path-layout.md

```diff
@@ -0,0 +1,255 @@
+# OCFL Community Extension NNNN: Direct Clean Path Layout
+
+* **Extension Name:** NNNN-direct-clean-path-layout
+* **Authors:** Jürgen Enge, Basel
+* **Minimum OCFL Version:** 1.0
+* **OCFL Community Extensions Version:** 1.0
+* **Obsoletes:** n/a
+* **Obsoleted by:** n/a
+
+## Overview
+
+This extension can be used as root storage layout extension to map OCFL object
+identifiers to path names as well as a mapping for content path names.
+This is done by replacing or removing "dangerous characters" from names.
+
+The functionality has been derived from David A. Wheeler - "Fixing Unix/Linux/POSIX Filenames" (https://www.dwheeler.com/essays/fixing-unix-linux-filenames.html)
+
+#### Usage Scenario
+
+If you want to make sure, that the internal layout of the OCFL content 
+substructure or the object id is as close as possible to the original structure, but you want to 
+make sure, that there are no possibly dangerous file- or foldernames, which 
+may oppose with durability.
+
+One Example could be complete filesystems, which come from estates, and are not
+under control of the archivists.
+
+### Caveat
+
+#### Projection
+This extension does  provide injective (one-to-one) projections only if 
+`encodeUTF == true`.  
+If you don't want to use the longish UTF-Code replacements, first check your source to make sure, that there's no chance of 
+different names to be mapped on one target. Software which generates the OCFL 
+structure should raise an error in this case.
+
+#### Object Identifiers
+Besides the caveat below, when using as storage layout root extension, you must 
+be sure, that no object identifier is  contained in another identifier as a 
+prefix path.  
+I.e. https://hdl.handle.net/XXXXX/test and https://hdl.handle.net/XXXXX/test/blah 
+could result to into storage root folders like  
+https/hdl.handle.net/XXXXX/test    
+https/hdl.handle.net/XXXXX/test/blah  
+This is an invalid OCFL storage root hierarchy.
+
+#### Length
+Several filesystems (e.g. Ext2/3/4) have byte restrictions on filename length.
+When using UTF16 (i.e. NTFS) or UTF32 characters in the filesystem, the byte length is double or quad of the character length.
+Using UTF8 characters in filesystems, byte length can only be determined by checking each string for maximum.  
+**Hint:** UTF8 has always less or equal bytes than UTF32 which means, that 
+assuming UTF32 instead of UTF8 for length calculation is safe, but would give you
+only 63 characters on 255 byte restrictions. 
+
+## Parameters
+
+### Summary
+
+* **Name:** `encodeUTF`
+    * **Description:** decides whether "dangerous" characters will be replaced by a 
```

Comment:
For consistence, I would probably capitalize the first letter of the first word of all of the words that follow a "Description:" heading.